registry/storage/driver/s3-aws/s3.go

```diff
@@ -506,6 +579,13 @@ func (d *driver) Name() string {
 
 // GetContent retrieves the content stored at "path" as a []byte.
 func (d *driver) GetContent(ctx context.Context, path string) ([]byte, error) {
+	if d.LogS3APIRequests {
```

Comment:
Is there a reason to do this on each request, rather than adding the handler once on driver initialization?

Reply:
We need the `ctx` argument to get the logger.

Reply:
Although I suppose we could pass a logger as a driver parameter and just use the presence or absence of that to determine whether or not to log S3 requests.

Reply:
Only problem with passing a logger as a driver parameter if i understand correctly is that normally one passes S3 driver parameters through a configuration file, although it wouldn't be a problem for us since we have our own implementation of `distribution.Namespace` and `distribution.Repository` that allows us to programmatically create driver parameters.

Reply:
Ah, right, we do want to use the `ctx`'s logger, so this makes sense. My only worry is whether these driver methods can be called concurrently, since `PushBackNamed` and `RemoveByName` don't look thread-safe.